src/utils/types/type_helpers.py

```diff
@@ -16,6 +16,22 @@ def is_list_of(obj: Any, query_type: type[T]) -> TypeGuard[list[T]]:
     return isinstance(obj, list) and all(isinstance(entry, query_type) for entry in obj)
 
 
+Predicate: TypeAlias = Callable[[T], bool]
+
+
+def all_true(predicates: list[Predicate[T]]) -> Predicate[T]:
+    """Compose a list of predicates into a single predicate.
+
+    The composed predicate will return True if any of the predicates in the list
```

Comment:
I think this should be
"...will return true if *all of the predicates in the list ...."

Reply:
Ah good spotting! That was originally what the function did, until I realised it wasn't what I needed. I'll fix it